### ISSUE TITLE
Missing double quotes for annotation values in Istio helm chart gateways template service.yaml file

### ISSUE THREAD

**Issue by PandaXass** [istio/istio#11851]
**Describe the bug**
The `serviceAnnotations` is not interpolated correctly in `service.yaml` file under gateways templates.

**Expected behavior**
The annotation values shall be double quoted.

**Steps to reproduce the bug**
Create a helm values yaml file with following configurations:
```yaml
gateways:
  istio-ingressgateway:
    serviceAnnotations:
      foo: "bar"
```
And then run `helm template install/kubernetes/helm/istio --name istio --namespace istio-system > $HOME/istio.yaml` according to istio setup documentation.
In the generated `istio.yaml` file, you can see the annotation value `bar` is not double quoted. And this will result some annotations not working properly.

**Version**
```
$ istioctl version
Version: 1.0.6
GitRevision: 98598f88f6ee9c1e6b3f03b652d8e0e3cd114fa2
User: root@464fc845-2bf8-11e9-b805-0a580a2c0506
Hub: docker.io/istio
GolangVersion: go1.10.4
BuildStatus: Clean

$ kubectl version
Client Version: version.Info{Major:"1", Minor:"12", GitVersion:"v1.12.2", GitCommit:"17c77c7898218073f14c8d573582e8d2313dc740", GitTreeState:"clean", BuildDate:"2018-10-30T21:39:38Z", GoVersion:"go1.11.1", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"11", GitVersion:"v1.11.7", GitCommit:"65ecaf0671341311ce6aea0edab46ee69f65d59e", GitTreeState:"clean", BuildDate:"2019-01-24T19:22:45Z", GoVersion:"go1.10.7", Compiler:"gc", Platform:"linux/amd64"}
```

**Installation**
Istio was installed according to https://istio.io/docs/setup/kubernetes/helm-install/#installation-steps option 1.

**Environment**
Environment: MacOS High Sierra
Cloud vendor: Azure AKS
OS: Ubuntu 16.04.5 LTS

**Comment by esnible** [comment]
Have you verified that some annotations don't work?

YaML is very tolerant of quotes.  See https://stackoverflow.com/questions/19109912/do-i-need-quotes-for-strings-in-yaml .  Please supply an example that breaks because of missing quotes.

**Comment by PandaXass** [comment]
Hi @esnible,

Thank you for your quick reply!

Here is a value yaml file example which does not work:
```yaml
gateways:
  istio-ingressgateway:
    serviceAnnotations:
      service.beta.kubernetes.io/azure-load-balancer-internal: "true"
    type: LoadBalancer
```
In this case, in Azure, a public load balancer would be created.
After I added double quotes in chart template file, it worked as expected.

**Comment by guitmz** [comment]
looks like this is fixed in the v1.1.X chart but not in v1.0.X chart

**Comment by PandaXass** [comment]
@guitmz yes, it is, I will close this issue :)
